toc/rfc/app-runtime-interfaces/rfc-draft-user-creation-by-org-managers.md

```diff
@@ -0,0 +1,63 @@
+# Meta
+[meta]: #meta
+- Name: User Creation by Org Managers
+- Start Date: 2024-08-13
+- Author(s): stephanme
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: [#946](https://github.com/cloudfoundry/community/pull/946)
+
+
+## Summary
+
+Allow Org Managers to create users in UAA in order to improve the onboarding procedure for larger developer groups into multi-tenant Cloud Foundry foundations.
+
+## Problem
+
+In order to assign CF roles to users, the users must be available in the UAA. As of today, the user is created in UAA explicitly by an administrator (origin=uaa) or implicitly when the user authenticates the first time via CF UAA for setups where an external identity provider is used (other origins).
+
+This is an issue for customers that want to onboard larger developer groups into multi-tenant Cloud Foundry foundations. All developers would have to log-in into the foundation before CF roles can be assigned - which is impractical. Customers should be able to automate the complete onboarding e.g. using CF CLI or other CF API clients like a CF terraform provider.
+
+## Proposal
+
+The proposal is to grant Org Managers the right to assign CF roles to users which are not yet known in UAA. The users are created in UAA either explicitly via an enhanced `POST /v3/users` call or implicitly when an Org Manager assigns an organization role using `POST /v3/roles`.
+
+The functionality SHALL be gated by a feature flag and SHALL be disabled by default.
+
+## Required Changes by Component
+
+### Cloud Controller
+
+#### Feature Flags
+
+Introduce a new [feature flag](https://v3-apidocs.cloudfoundry.org/version/3.172.0/index.html#list-of-feature-flags) `user_creation_by_org_manager` which is disabled by default .
```

Comment:
Should this be an API feature flag, or a configuration property in the deployment manifest? Would this be something that CC admins would want to toggle at run time?

Reply:
I chose a feature flag at first because of the related feature flag `set_roles_by_username`.

But indeed, a deploy time config property should be sufficient. I can't imagine of an use case where CC admins have to change this at run time.